src/main/scala/akka/persistence/jdbc/query/JournalSequenceActor.scala

```diff
@@ -117,14 +132,12 @@ class JournalSequenceActor(readJournalDao: ReadJournalDao, config: JournalSequen
 
             // if it's a gap and has been detected before on a previous iteration we give up
             // that means that we consider it a genuine gap that will never be filled
-            case e if givenUp(e)               => missing
```

Comment:
IIUC, there was a bug here. Because `e` was never added to the missing Seq, this check would never return true and we would always add the gap in the next case (if not already missing).